instrumentation/log4j/log4j-appender-2.17/library/src/main/java/io/opentelemetry/instrumentation/log4j/appender/v2_17/OpenTelemetryAppender.java

```diff
@@ -187,10 +214,30 @@ private static List<String> splitAndFilterBlanksAndNulls(String value) {
    */
   public void setOpenTelemetry(OpenTelemetry openTelemetry) {
     this.openTelemetry = openTelemetry;
+    LogEventToReplay eventToReplay;
+    while ((eventToReplay = eventsToReplay.poll()) != null) {
+      append(eventToReplay);
+    }
   }
 
+  @SuppressWarnings("SystemOut")
   @Override
   public void append(LogEvent event) {
+    // TODO(jean): Race condition to fix
+    // time=1 append() thread gets the no-op instance
+    // time=2 install() thread updates the instance and flushes the queue
+    // time=3 append() thread adds the log event to the queue
+    if (openTelemetry == OpenTelemetry.noop()) {
```

Comment:
Nit: you could store `this.openTelemetry` as a local variable and get rid of one extra volatile read later on